model/src/main/java/com/javahelp/model/user/UserInfo.java

```diff
@@ -33,6 +36,33 @@ public void setEmailAddress(String emailAddress) {
         this.emailAddress = emailAddress;
     }
 
+    /**
```

Comment:
Should these be moved to `ClientUserInfo` instead?

Reply:
ProviderUserInfo also needs the setEmailAddress method. If you look at the constructor for ProviderUserInfo, emailAddress is added to it as a private variable too.

Reply:
Oh the position of my message is displayed is a bit misunderstanding. I meant the one below, since both `searching` and `matching` are from Client UI.

Reply:
We'll have to get the attributeMaps of both Client and Providers to calculate similarities. I believe this is necessary for the ProviderRanker method.

Reply:
Alright gotcha

Reply:
If there are not other issues from your end, do remember to approve this PR